spec/controllers/v0/caregivers_assistance_claims_controller_spec.rb

```diff
@@ -64,38 +22,114 @@
+          expect(res_body['errors'][0]).to eq(
+            {
+              'title' => 'Missing parameter',
+              'detail' => 'The required parameter "caregivers_assistance_claim", is missing',
+              'code' => '108',
+              'status' => '400'
+            }
           )
+        end
+
+        it 'requires "caregivers_assistance_claim.form" param' do
+          expect_any_instance_of(Form1010cg::Service).not_to receive(:process_claim!)
 
-          post :create, params: { caregivers_assistance_claim: { form: expected[:form_data] } }
+          post :create, params: { caregivers_assistance_claim: { form: nil } }
 
-          expect(response).to have_http_status(:ok)
+          expect(response).to have_http_status(:bad_request)
 
-          res_body = JSON(response.body)
+          res_body = JSON.parse(response.body)
+
+          expect(res_body['errors'].size).to eq(1)
+          expect(res_body['errors'][0]).to eq(
+            {
+              'title' => 'Missing parameter',
+              'detail' => 'The required parameter "form", is missing',
+              'code' => '108',
+              'status' => '400'
+            }
+          )
+        end
 
-          expect(res_body['data']).to be_present
-          expect(res_body['data']['id']).to eq('')
-          expect(res_body['data']['attributes']).to be_present
-          expect(res_body['data']['attributes']['confirmation_number']).to eq(expected[:carma_case_id])
-          expect(res_body['data']['attributes']['submitted_at']).to eq(expected[:submitted_at])
+        context 'when submission is' do
+          context 'invalid' do
+            it 'builds a claim and raises it\'s errors' do
+              params = { caregivers_assistance_claim: { form: '{}' } }
+              form_data = params[:caregivers_assistance_claim][:form]
+              claim = build(:caregivers_assistance_claim, form: form_data)
+
+              expect(SavedClaim::CaregiversAssistanceClaim).to receive(:new).with(
+                form: form_data
+              ).and_return(
+                claim
+              )
+
+              expect(Form1010cg::Service).not_to receive(:new).with(claim)
+
+              post :create, params: params
+
+              res_body = JSON.parse(response.body)
+
+              expect(response.status).to eq(422)
+
+              expect(res_body['errors']).to be_present
+              expect(res_body['errors'].size).to eq(2)
+              expect(res_body['errors'][0]['title']).to include("did not contain a required property of 'veteran'")
+              expect(res_body['errors'][0]['code']).to eq('100')
+              expect(res_body['errors'][0]['status']).to eq('422')
+              expect(res_body['errors'][1]['title']).to include(
+                "did not contain a required property of 'primaryCaregiver'"
+              )
+              expect(res_body['errors'][1]['code']).to eq('100')
+              expect(res_body['errors'][1]['status']).to eq('422')
```

Comment:
We've now got the super_diff gem in the app -- it may or may not let you condense this a bit if you're interested in taking a look.

Reply:
This is tricky because I have to use an RSpec `include` matcher for this objects `title` property (because of the uuid present in the string).

I'm not seeing an example or spec in super_diff that allows you to use a Rspec `include` inside an `eq` matcher like so:
```rb
expect(res_body['errors'][0]).to eq(
  'title' => include("did not contain a required property of 'veteran'"),
  'code' => '100',
  'status' => '422'
 )
```